homeassistant/components/cover/group.py

```diff
@@ -0,0 +1,269 @@
+                self.tilts[KEY_POSITION].add(entity_id)
+                self._tilt = True
+
+    @asyncio.coroutine
+    def async_added_to_hass(self):
+        """Register callbacks."""
+        @callback
+        def state_change_listener(entity, old_state, new_state):
+            """Handle cover state changes and update state."""
+            self.update_supported_features(entity, old_state, new_state)
+            self.async_schedule_update_ha_state(True)
+
+        for entity_id in self.entities:
+            new_state = self.hass.states.get(entity_id)
+            self.update_supported_features(entity_id, None, new_state)
+        async_track_state_change(self.hass, self.entities,
+                                 state_change_listener)
+
+    @property
+    def name(self):
+        """Return the name of the cover."""
+        return self._name
+
+    @property
+    def assumed_state(self):
+        """Enable buttons even if at end position."""
+        return True
+
+    @property
+    def should_poll(self):
+        """Disable polling for the MultiCover."""
+        return False
+
+    @property
+    def is_closed(self):
+        """Return if all covers in group are closed."""
+        if not self.covers[KEY_OPEN_CLOSE]:
+            return False
+        for entity_id in self.covers[KEY_OPEN_CLOSE]:
+            state = self.hass.states.get(entity_id)
+            if state is None or state.state != STATE_CLOSED:
+                return False
+        return True
+
+    @property
+    def current_cover_position(self):
+        """Return current position for all covers.
+
+        None is unknown. pos if all have the same position.
+        Else 0 if closed or 100 is fully open.
+        """
+        position = -1
+        for entity_id in self.covers[KEY_POSITION]:
+            state = self.hass.states.get(entity_id)
+            pos = state.attributes.get(ATTR_CURRENT_POSITION)
+            if position == -1:
+                position = pos
+            elif position != pos:
+                position = -1
+                break
+
+        if position != -1:
+            return position
+
+        return 0 if self.is_closed else 100
+
+    @property
+    def current_cover_tilt_position(self):
+        """Return current tilt position for all covers.
+
+        None is unknown. pos if all have the same tilt position.
+        Else 100 for tilt open.
+        """
+        if self._tilt is False:
+            return None
+
+        position = -1
+        for entity_id in self.tilts[KEY_POSITION]:
+            state = self.hass.states.get(entity_id)
+            pos = state.attributes.get(ATTR_CURRENT_TILT_POSITION)
+            if position == -1:
+                position = pos
+            elif position != pos:
+                position = -1
+                break
+
+        if position != -1:
+            return position
+
+        return 100
+
+    @property
+    def state_attributes(self):
```

Comment:
If I'm correct the super method does exactly the same:

https://github.com/home-assistant/home-assistant/blob/f8a0a0ba595a5ff53bc5d81810bb69abd6e3990c/homeassistant/components/cover/__init__.py#L223-L236

Reply:
Missed that, thanks :+1: